app/src/utils/utils.ts

```diff
@@ -179,6 +181,43 @@ export async function assertSufficientBalance(tokenAddress: string, requiredAmou
   return true;
 }
 
+/**
+ * @notice Await for a transaction to complete and watch for replacements
+ * @param call ()=>ContractTransaction A call to init the transaction
+ * @param stateRef Ref<hash> A pointer to update with the tx hash/state
+ * @returns Promise<ContractTransaction> A promise of the final ContractTransaction that replaces the initial call() response
+ */
+export async function watchTransaction(
+  call: () => Promise<ContractTransaction>,
+  stateRef: Ref
+): Promise<ContractTransaction> {
+  // tx is picked up from the provided call and updated if replaced
+  let tx: ContractTransaction;
+  // do the call inside a try catch so that we see the error first
+  try {
+    // get the ContractTransaction from the call
+    tx = await call();
+    // provisionally set the tx.hash to move onto next page
+    stateRef.value = tx.hash;
+    // wait for the transaction to be mined
+    void (await tx.wait());
+  } catch (error) {
+    if (error.code === Logger.errors.TRANSACTION_REPLACED) {
```

Comment:
To address above something similar to this

Reply:
This is being drawn from ethers (https://docs.ethers.io/v5/api/utils/logger/#errors--transaction-replaced), I think its wallet agnostic, @mds1 any ideas here?